crates/turbopack-ecmascript/src/tree_shake/mod.rs

```diff
@@ -185,15 +180,14 @@ impl Analyzer<'_> {
 
                     if let Some(items) = items_to_remove_from_last_reads.get(id) {
                         for item in items {
-                            if let Some(pos) = state.last_reads.iter().position(|v| *v == *item) {
-                                state.last_reads.remove(pos);
-                            }
+                            state
+                                .last_reads
+                                .retain(|last_read| !self.g.has_strong_dep(item, last_read))
                         }
                     }
```

Comment:
``` rs
 // Optimization: Remove each module item to which we
// have a strong dependency.

state.last_reads.retain(|last_read| !self.g.has_strong_dep(item_id, last_read))
```

`items_to_remove_from_last_reads` is not needed here.

---

Same for `WRITE_VARS`.

We basically don't have to store A and B when there is a strong dependency from A to B. In that case we only need to store A, since B is transitively included in A.

Since we just added (or are about to add `item_id`, assuming we swap the statements, which we should do), we can remove all items on which `item_id` depends in a strong way.